src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_golemagg.cpp

```diff
@@ -58,13 +60,27 @@ class boss_golemagg : public CreatureScript
             earthquakeTimer = 0;
             pyroblastTimer = urand(3000, 7000);
             enraged = false;
+            DoCastSelf(SPELL_MAGMASPLASH);
+            DoCastSelf(SPELL_GOLEMAGG_TRUST_AURA);
+            DoCastSelf(SPELL_DOUBLE_ATTACK);
+
+            instance->SetBossState(DATA_GOLEMAGG, FAIL);
+        }
+
+        void JustDied(Unit* /*killer*/) override
```

Comment:
were the states not updating correctly before? because without the override, they should update by default

Reply:
Without this changes there were no updates before.

Reply:
sounds to me like a bug with BossAI and should perhaps be tackled there instead of inside every script?

Reply:
Idk why though, I've just checked BossAI and it must update the BossState. But adding this appears to solve the issue, unless there is another way

Reply:
@Kitzunu perhaps, it should be checked by someone

Reply:
how did you check that the states were not updating?

Reply:
Because this script was never executed (i put a breakpoint using debug to see if that script was working or not) https://github.com/azerothcore/azerothcore-wotlk/blob/707b6d1234e644391dd4589991f1031b0b46f54b/src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp#L268

Reply:
I just tried and it's working for me without the override. Had you killed him before you tried debugging it? If it's set to DONE the state doesn't get updated again

https://user-images.githubusercontent.com/47818697/142301572-c0c85bf0-6cc8-4557-b485-918eefacdde3.mp4

Reply:
Odd, I tried multiple times and it didn't get executed, and it seems it happened to the report author too. I'll try again when I come back home.

Reply:
Yep, I tried again and now it worked for me. I'm reverting the setboss changes.